crates/node/src/catchup.rs

```diff
@@ -2,48 +2,96 @@ use std::collections::VecDeque;
 use libp2p::PeerId;
 use rand::seq::SliceRandom;
 use rand::thread_rng;
 use serde_json::{from_slice as from_json_slice, to_vec as to_json_vec};
+use tokio::spawn;
+use tokio::sync::mpsc;
 use tokio::time::timeout;
 use tracing::{error, info, warn};
+use url::Url;
 
-use crate::catchup::batch::CatchupBatchSender;
+use crate::catchup::blobs::{ApplicationBlobChunkSender, ApplicationBlobChunkStream};
+use crate::catchup::transactions::TransactionsBatchSender;
 use crate::transaction_pool::TransactionPoolEntry;
 use crate::types::{
-    CatchupError, CatchupStreamMessage, CatchupTransactionsRequest, TransactionStatus,
-    TransactionWithStatus,
+    CatchupApplicationBlobRequest, CatchupError, CatchupStreamMessage, CatchupTransactionsBatch,
+    CatchupTransactionsRequest, TransactionStatus, TransactionWithStatus,
 };
 use crate::Node;
 
-mod batch;
+mod blobs;
+mod transactions;
 
 impl Node {
-    // TODO: Consider splitting this long function into multiple parts.
-    #[expect(clippy::too_many_lines, reason = "TODO: Will be refactored")]
     pub(crate) async fn handle_opened_stream(&self, mut stream: Box<Stream>) -> EyreResult<()> {
         let Some(message) = stream.next().await else {
             bail!("Stream closed unexpectedly")
         };
 
-        let request = match from_json_slice(&message?.data)? {
-            CatchupStreamMessage::TransactionsRequest(req) => req,
+        match from_json_slice(&message?.data)? {
+            CatchupStreamMessage::TransactionsRequest(req) => {
+                self.handle_transaction_catchup(req, stream).await
+            }
+            CatchupStreamMessage::ApplicationBlobRequest(req) => {
+                self.handle_blob_catchup(req, stream).await
+            }
             message @ (CatchupStreamMessage::TransactionsBatch(_)
-            | CatchupStreamMessage::ApplicationBlobRequest(_)
             | CatchupStreamMessage::ApplicationBlobChunk(_)
             | CatchupStreamMessage::Error(_)) => {
                 bail!("Unexpected message: {:?}", message)
             }
+        }
+    }
+
+    async fn handle_blob_catchup(
+        &self,
+        request: CatchupApplicationBlobRequest,
+        mut stream: Box<Stream>,
+    ) -> Result<(), eyre::Error> {
+        let Some(mut blob) = self
+            .ctx_manager
+            .get_application_blob(&request.application_id)?
+        else {
+            let message = to_json_vec(&CatchupStreamMessage::Error(
+                CatchupError::ApplicationNotFound {
+                    application_id: request.application_id,
+                },
+            ))?;
+            stream.send(Message::new(message)).await?;
+
+            return Ok(());
         };
 
+        // Stream messages are encoded with length delimited codec.
+        // Max message size is 8MB and blob chunk size 1MB (atm).
+        // So, collect 7 chunks and push them to over the stream.
+        let mut blob_sender = ApplicationBlobChunkSender::new(7, stream);
```

Comment:
can we `const`antly make this calculation? in case either of its dependencies change

(MAX_MESSAGE_SIZE / BLOB_CHUNK_SIZE ) - 1